### PR TITLE
[init][draft] trino lance connector boiler plate code

### PR DESCRIPTION
**Pull request by walterddr**

### CODE REVIEW

File: plugin/trino-lancedb/src/main/java/io/trino/plugin/lance/internal/LanceReader.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        dbScheme = lanceDbURI.getScheme();
+        dbPath = lanceDbURI.getPath();
+    }
+
+    private static Schema getSchema(String tablePath) {
+        try (Dataset dataset = Dataset.open(tablePath, allocator)) {
+            Scanner scanner = dataset.newScan();
+            return scanner.schema();
+        }
+    }
```
Comment (walterddr) [inline plugin/trino-lancedb/src/main/java/io/trino/plugin/lance/internal/LanceReader.java]: would be nice if i dont have to create the Dataset in order to get the schema

File: plugin/trino-lancedb/src/main/java/io/trino/plugin/lance/internal/LanceReader.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+    private static String getTablePath(String dbPath, String tableName) {
+        // TODO: local fs impl here to be replaced by
+        // https://lancedb.github.io/lancedb/python/python/#lancedb.db.DBConnection.open_table
```
Comment (walterddr) [inline plugin/trino-lancedb/src/main/java/io/trino/plugin/lance/internal/LanceReader.java]: should be replace with open_table

File: plugin/trino-lancedb/src/main/java/io/trino/plugin/lance/internal/LanceReader.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        Path tablePath = Paths.get(dbPath, tableName);
+        return tablePath.toString();
+    }
+
+    public List<SchemaTableName> listTables(ConnectorSession session, String schema) {
+        // TODO: local fs impl here to be replaced by
+        // https://lancedb.github.io/lancedb/python/python/#lancedb.db.DBConnection.table_names
```
Comment (walterddr) [inline plugin/trino-lancedb/src/main/java/io/trino/plugin/lance/internal/LanceReader.java:70]: should be replace with table_names

File: plugin/trino-lancedb/src/test/java/io/trino/plugin/lance/TestLanceMetadata.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_METHOD;
+
+
+@TestInstance(PER_METHOD)
+public class TestLanceMetadata
+{
+    private static final LanceTableHandle TEST_TABLE_1_HANDLE = new LanceTableHandle("example", "test_table1");
+    private static final LanceTableHandle TEST_TABLE_2_HANDLE = new LanceTableHandle("example", "test_table2");
```
Comment (rong-cai) [inline plugin/trino-lancedb/src/test/java/io/trino/plugin/lance/TestLanceMetadata.java]: currently i only have 1 test table copied from lance. would be nice if we have more with various data schema. please advice if there's any existing test resource i can copy over

Comment (eddyxu) [reply plugin/trino-lancedb/src/test/java/io/trino/plugin/lance/TestLanceMetadata.java]: https://github.com/lancedb/lance/tree/main/test_data We have test dataset.

File: plugin/trino-lancedb/src/main/java/io/trino/plugin/lance/internal/LanceArrowToPageScanner.java [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+        vectorSchemaRoot.clear();
+    }
+
+    private void convertType(BlockBuilder output, Type type, FieldVector vector, int offset, int length) {
+        Class<?> javaType = type.getJavaType();
+        try {
+            if (javaType == boolean.class) {
+                writeVectorValues(output, vector,
+                        index -> type.writeBoolean(output, ((BitVector) vector).get(index) == 1), offset, length);
+            } else if (javaType == long.class) {
```
Comment (eddyxu) [inline plugin/trino-lancedb/src/main/java/io/trino/plugin/lance/internal/LanceArrowToPageScanner.java]: How does Trino present "Vector"?

In lance, a vector is `arrow.FixedSizeList<float16/float32/float64>`

Comment (walterddr) [reply plugin/trino-lancedb/src/main/java/io/trino/plugin/lance/internal/LanceArrowToPageScanner.java]: Array i think

File: plugin/trino-lancedb/src/test/java/io/trino/plugin/lance/TestLanceDatasetPageSource.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        assertThat(lanceDbURL)
+                .describedAs("example db is null")
+                .isNotNull();
+        LanceConfig lanceConfig = new LanceConfig().setLanceDbUri(lanceDbURL.toString());
+        LanceReader lanceReader = new LanceReader(lanceConfig);
+        metadata = new LanceMetadata(lanceReader, lanceConfig);
+    }
+
+    @Test
+    public void testTableScan()
```
Comment (walterddr) [inline plugin/trino-lancedb/src/test/java/io/trino/plugin/lance/TestLanceDatasetPageSource.java:39]: this test failed with the following command

```
mvn install -Dair.check.skip-all -pl plugin/trino-lancedb
mvn test -Dair.check.skip-all -pl plugin/trino-lancedb -Dtest=TestLanceDatasetPageSource*
```

the failure is relatively hard to debug but i can't be sure what's wrong. we can probably first walk through the test case and see what's wrong first. and next we can chat about what's the proper way to deal with it. 

problem with running in intellij is that it attaches self during java debugging and thus making a AttachMain <--> JUnitStarter process dependency loop. and i cant kill them.

running with mvn is good but i cant debug what's wrong.

Comment (walterddr) [reply plugin/trino-lancedb/src/test/java/io/trino/plugin/lance/TestLanceDatasetPageSource.java:39]: CC @eddyxu

**Comment by walterddr** [comment]
first version runnable closing and move to upstream

**Pull request closed without merging**